Added a new JITM message for Manage on the update-core.php admin page

jessefriedman opened the pull request:
This JITM is designed to guide users to use WordPress.com tools. The message will appear (with variations to the call-to-action) whether Jetpack Manage is active or not. 

If Jetpack Manage is not active the user can activate it through an ajax call in the message. Once Manage is active the call-to-action becomes a dynamic link to that specific sites plugin management page on WordPress.com.

How to test:
- Checkout branch, run grunt!
- Start with both Photon and Mange not active/enabled
- Head over to the Updates page in the admin (wp-admin/update-core.php)
- View the message make sure it looks good, click the "Activate WordPress.com Tools" button
- You should receive a success message and a new button saying "Go to WordPress.com" which should take you to your plugins page for the site you're working in
- Go back to the admin and refresh the update-core.php page. The message should not disappear. 
- Dismissing the notice (x top right corner) should make the JITM disappear forever (only reset Jetpack options will bring it back)
Final Test
- Head over to Media -> Add New and make sure the Photon JITM works/looks good. That notice should be present if Photon is not enabled and it's never been dismissed. Dismissing one JITM doesn't dismiss all JITM's.

zinigor commented:
@jessefriedman @jeffgolenski and anyone who works with this branch - please beware, I have force-pushed it, please delete your local branches before pulling.

zinigor commented:
I have found some problems while testing this:
* on the updates page when clicking the "Activate" button it appears as though nothing is happening, perhaps there should be a spinner or something; the button is not disabled when clicking;
* the JITM message button has a wrong vertical alignment, or so it seems: ![screen shot 2015-10-03 at 00 59 12](https://cloud.githubusercontent.com/assets/374293/10258798/28564932-696a-11e5-857a-254bc123e556.png)

Everything else works as expected!

jessefriedman commented:
@zinigor I have updated the PR with a spinner and fixed the alignment issue. I didn't notice the lag when waiting on the ajax before but I'm glad you pointed it out.

dereksmart commented on class.jetpack-jitm.php:
```diff
@@ -21,12 +21,53 @@ static function init() {
 	}
 
 	private function __construct() {
+		if ( ! Jetpack::is_active() ) {
```
Might make sense to add an early check for `current_user_can( 'jetpack_manage_modules' )` as well.

jessefriedman replied:
while we call the `current_user_can()` function after `capabilities.php` has loaded `wp_get_current_user()` is not yet available, so this will result in an error. There may be a way to hook into it if we require the file

dereksmart replied:
Not really worth it IMO

dereksmart commented on class.jetpack-jitm.php:
```diff
@@ -21,12 +21,53 @@ static function init() {
 			add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
 			add_action( 'post-plupload-upload-ui', array( $this, 'photon_msg' ) );
 		}
+		else if ( 'update-core.php' == $pagenow && 'hide' != $showmanage ) {
```
Let's also check if manage is enabled here before we do anything

dereksmart replied:
jk, I forgot that there's a second CTA

dereksmart commented on class.jetpack-jitm.php:
```diff
@@ -21,12 +21,53 @@ static function init() {
+	function manage_msg() {
+		if ( current_user_can( 'jetpack_manage_modules' ) ) {
+			$normalized_site_url = Jetpack::build_raw_urls( get_home_url() );
+			$manage_active = Jetpack::is_module_active( 'manage' );
```
Move this to my comment above

dereksmart replied:
JK about ^ comment

dereksmart commented:
Alright this looks reasonable to me.  @zinigor when you get up there is some discussion about it in our slack channel.  I'll leave you to merge it if looks ok to you.

zinigor commented on _inc/jetpack-jitm.js:
```diff
@@ -29,43 +30,48 @@
-		$('.jp-jitm .activate').click(function() {
+		$( '.jp-jitm .activate' ).click(function() {
 
+			$( '.button').addClass( 'hide' );
```
Tiny nitpick: space after `'.button'`.

zinigor commented on _inc/jetpack-jitm.js:
```diff
@@ -29,43 +30,48 @@
-					}, 5000);
+					$( '.msg' ).html( success_msg );
+					$( '#jetpack-wordpressdotcom, .jetpack-spinner' ).toggleClass( 'hide' );
+					if ( 'manage' !== data.jitmModule ){
```
Another small nitpick: space before curly brace.

zinigor commented on _inc/jetpack-jitm.js:
```diff
@@ -29,43 +30,48 @@
+					}
 				} else {
-					$('.jp-jitm').html('<p><span class="icon"></span>'+fail_msg+'</p>');
+					$( '.jp-jitm' ).html( '<p><span class="icon"></span>'+fail_msg+'</p>' );
```
Yet another one: spaces around `+` signs.

zinigor commented:
This looks good except for a few minor nitpicks, I'm going to fix them myself and rebase the PR in ~30 minutes.
